src/tmt_web/api.py

```diff
@@ -1,29 +1,70 @@
+import logging
 import os
-from typing import Annotated, Any, Literal
+import platform
+import time
+from datetime import UTC, datetime
+from typing import Annotated, Literal
 
 from celery.result import AsyncResult
-from fastapi import FastAPI
+from fastapi import FastAPI, Request, status
 from fastapi.params import Query
+from fastapi.responses import HTMLResponse, JSONResponse, PlainTextResponse, RedirectResponse
 from pydantic import BaseModel
-from starlette.responses import HTMLResponse, JSONResponse, PlainTextResponse
+from tmt import Logger
+from tmt import __version__ as tmt_version
+from tmt.utils import GeneralError
 
 from tmt_web import service, settings
 from tmt_web.generators import html_generator
 
-app = FastAPI()
+# Record start time for uptime calculation
+START_TIME = time.time()
+
+# Create main logger for the API
+logger = Logger(logging.getLogger("tmt-web-api"))
+
+app = FastAPI(
+    title="tmt Web API",
+    description="Web API for checking tmt tests, plans and stories",
+    version="1.0.0",
+)
 
 
 class TaskOut(BaseModel):
+    """Response model for asynchronous tasks."""
+
     id: str
     status: str
     result: str | None = None
     status_callback_url: str | None = None
 
 
+class HealthStatus(BaseModel):
+    """Health check response model."""
+
+    status: str
+    timestamp: datetime
+    uptime_seconds: float
+    version: dict[str, str]
+    dependencies: dict[str, str]
+    system: dict[str, str]
```

Comment:
`version`, `dependencies`, and `system` dicts could get their model as well, asuming the dict keys are constant.

Reply:
Like this? 37f0c726ca1c1e98c2a508a520b93b7043e753c5